registry/storage/manifeststore.go

```diff
@@ -154,7 +154,55 @@ func (ms *manifestStore) Put(ctx context.Context, manifest distribution.Manifest
 // Delete removes the revision of the specified manifest.
 func (ms *manifestStore) Delete(ctx context.Context, dgst digest.Digest) error {
 	dcontext.GetLogger(ms.ctx).Debug("(*manifestStore).Delete")
-	return ms.blobStore.Delete(ctx, dgst)
+
```

Comment:
What's below is essentially the body of `ms.blobStore.Delete(ctx, dgst)`, expanded with referrers deletion operations.

Reply:
We should not expand `ms.blobStore.Delete(ctx, dgst)`. Instead, can we reuse `ms.blobStore.Delete(ctx, dgst)`, `ms.blobStore.Stat()`, and `ms.Get()`?

Reply:
Changed accordingly.